src/app/mod.rs

```diff
@@ -47,7 +49,7 @@ pub struct CeleryBuilder {
 
 impl CeleryBuilder {
     /// Get a [`CeleryBuilder`] for creating a [`Celery`] app with a custom configuration.
-    pub fn new(name: &str, broker_url: &str) -> Self {
+    pub fn new(name: &str, broker_url: &str, backend_url: Option<impl AsRef<str>>) -> Self {
```

Comment:
> we just add a method for setting this

This is less straight forward. Given that there are other options for the backend which modify the backend builder we would then need either a type parameter for the entire `CeleryBuilder` so that those setters are only exposed once you have set the backend url and the backend builder is present or we need to store the properties in the `CeleryBuilder` and then only use them when a backend_url was actually specified.

The first option is a breaking change as well and the second option not great from a usability point of view.

In any case, the changes in this MR are breaking the existing API in any case as `AsyncResult` is now `AsyncResult<T>`. The changes around `TaskError` and the requirement for `Task::Returns: 'static` are also breaking. As would be the change of `&str` -> `AsRef<str>` for `broker_url`.

Happy to make the adjustments you want, let me know which route you prefer.

Reply:
@epwalsh @rksm has this PR ended up dead? If so, I think that would be a same as it appears quite valuable for the usability of the framework.